app/src/main/java/org/wildaid/ofish/ui/reportdetail/ReportDetailFragment.kt

```diff
@@ -76,6 +77,9 @@ class ReportDetailFragment : Fragment(R.layout.fragment_report_details) {
 
         report_toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white)
         subscribeToDialogEvents()
+
+        val shouldShowButton = requireArguments().getSerializable(KEY_SHOULD_SHOW_BUTTON) as Boolean
```

Comment:
Use get boolean from arguements.
Now Application crashes ig user navigates here from Vessel details fragment. Get boolean with default value will resolve it